v2/astradb-to-bigquery/src/main/java/com/google/cloud/teleport/v2/astradb/templates/AstraDbToBigQuery.java

```diff
@@ -0,0 +1,216 @@
+       * is optional. If not set, the table specification is built from the cassandra source table
+       * attributes: keyspace=dataset name, table=table name.
+       */
+      String bigQueryOutputTableSpec = astra2BqOptions.getOutputTableSpec();
+      if (StringUtils.isEmpty(bigQueryOutputTableSpec)) {
+        bigQueryOutputTableSpec =
+            astra2BqOptions.getProject()
+                + ":"
+                + astra2BqOptions.getAstraKeyspace()
+                + "."
+                + astra2BqOptions.getAstraTable();
+      }
+      TableReference bigQueryTableReference =
+          BigQueryUtils.toTableReference(bigQueryOutputTableSpec);
+      LOGGER.info("Big Query table spec has been set to {}", bigQueryOutputTableSpec);
+      /*
+       * If the destination dataset does not exist, it is created before running pipeline.
+       */
+      BigQuery bigquery =
+          BigQueryOptions.newBuilder()
+              .setProjectId(astra2BqOptions.getProject())
+              .build()
+              .getService();
+      if (null
+          == bigquery.getDataset(
+              DatasetId.of(
+                  bigQueryTableReference.getProjectId(), bigQueryTableReference.getDatasetId()))) {
+        LOGGER.info(
+            "Dataset was not found: creating DataSet {} in region {}",
+            bigQueryTableReference.getDatasetId(),
+            astra2BqOptions.getWorkerRegion());
+        bigquery.create(
+            DatasetInfo.newBuilder(bigQueryTableReference.getDatasetId())
+                .setLocation(astra2BqOptions.getWorkerRegion())
+                .build());
+        LOGGER.debug("Dataset has been created [OK]");
+      }
+
+      /*
+       * Generic Serializer Cassandra Record => Beam Row.
+       *
+       * - Most types are supported. UDT and Tuple are not.
+       * - List, Set are converted to ARRAY
+       * - Columns part of the primary are not REQUIRED.
+       * - Mapper is created based on the source cassandra table schema.
+       */
+      SerializableFunction<Session, Mapper> beamRowMapperFactory =
+          new BeamRowObjectMapperFactory(
+              astra2BqOptions.getAstraKeyspace(), astra2BqOptions.getAstraTable());
+
+      /*
+       * MAPPING #2: Mapper Cassandra Table => BigQuery Schema.
+       */
+      SerializableFunction<AstraDbIO.Read<?>, TableSchema> bigQuerySchemaFactory =
+          new AstraDbToBigQuerySchemaMapperFn(
+              astra2BqOptions.getAstraKeyspace(), astra2BqOptions.getAstraTable());
+      LOGGER.debug("Serializer initializations [OK]");
+
+      // Source: AstraDb
+      AstraDbIO.Read<Row> astraSource =
+          AstraDbIO.<Row>read()
+              .withToken(astraToken)
+              .withSecureConnectBundleData(astraSecureBundle)
+              .withKeyspace(astra2BqOptions.getAstraKeyspace())
+              .withTable(astra2BqOptions.getAstraTable())
+              .withMinNumberOfSplits(5)
```

Comment:
Does it make sense to hardcode? Maybe add an optional parameter?

Reply:
The latter version has externalized this parameter